app/models/user.rb

```diff
@@ -6,7 +6,41 @@ class User < ApplicationRecord
 
   validates :name, presence: true, length: { maximum: 20 }
 
-  has_many :posts
+  has_many :posts, dependent: :destroy
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  has_many :friendships, dependent: :destroy
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: :friend_id, dependent: :destroy
+  # Added these for the other options for user status
+  has_many :accepted_friendships, -> { where(status: true) }, class_name: 'Friendship'
+  has_many :pending_friendships, -> { where(status: false) }, class_name: 'Friendship', foreign_key: :friend_id
+
+  # enum(:confirmed, :pending_friends, :requested_friends)
+  # status = confirmed and status == 'confirmed'
+  def friends
+    friends_array = friendships.map { |friendship| friendship.friend if friendship.status == true }
+    friends_array += inverse_friendships.map { |friendship| friendship.user if friendship.status }
+    friends_array.compact
+  end
+
+  # Users who have yet to confirmed friend requests
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.status }.compact
+  end
+
+  # Users who have requested to be friends
```

Comment:
It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
See this [post](https://stackoverflow.com/questions/13130859/inline-comments-in-ruby/13130900#:~:text=No%2C%20Ruby%20does%20not%20have,comment%20out%20the%20one%20row), about why comments are discouraged.